testproject/Assets/Scripts/Raygun.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Raygun : MonoBehaviour
+{
+  public float range = 15;
+
+  private GameObject currentTarget;
+  private Rigidbody rb;
+
+  void Start()
+  {
+    StartCoroutine(Attack());
+    rb = GetComponent<Rigidbody>();
+  }
+  void Update()
+  {
+    Vector3 forward = transform.TransformDirection(Vector3.forward) * range;
+    Debug.DrawRay(transform.position, forward, Color.yellow);
+  }
+
+  IEnumerator Attack()
+  {
+    currentTarget = FindTarget();
+
+    ShootTarget();
+
+    yield return new WaitForSeconds(10);
```

Comment:
Just wondering if starting a new Coroutine for each iteration is better than:
```
bool m_CanAttack;

private void OnEnable()
{
	m_CanAttack = true;
	StartCoroutine(Attack());
}

private void OnDisable()
{
	m_CanAttack = false;
	StopCoroutine(Attack());
}

IEnumerator Attack()
{
	while(m_CanAttack)
	{
		currentTarget = FindTarget();
		ShootTarget();
		yield return new WaitForSeconds(10);
	}
}
```

Reply:
I think a prefer kicking off the Attack at the end of each Attack,, but it could lead to some crazy stack traces. I guess my question would be what's the more 'Unity' style of doing this?